### PR TITLE
[#3808] flag ppport.h files created by Devel::PPPort perl module as generated

### PR DESCRIPTION
**Pull request by ioanrogers**
Marking `ppport.h` files which are generated by [Devel::PPPort](https://metacpan.org/pod/Devel::PPPort) as generated.

## Checklist:

- [x] **I am adding new or changing current functionality**
  
  - [x] I have added or updated the tests for the new or changed functionality.

### CODE REVIEW

File: lib/linguist/generated.rb [hunk 1]
```diff
@@ -533,5 +534,14 @@ def generated_grpc_cpp?
       return false unless lines.count > 1
       return lines[0].start_with?("// Generated by the gRPC")
     end
+
+    # Internal: Is the file a generated Perl/Pollution/Portability header file?
+    #
+    # Returns true or false.
+    def generated_perl_ppport_header?
+        return false unless name.match(/ppport\.h$/)
+        return false unless lines.count > 10
```
Comment (pchaigno) [inline lib/linguist/generated.rb]: Why 10? Is that arbitrary?

Comment (ioanrogers) [reply lib/linguist/generated.rb]: The string match in the next test is on line 9, so it's just to make sure the file is at least that long.

File: lib/linguist/generated.rb [hunk 2]
```diff
@@ -533,5 +534,14 @@ def generated_grpc_cpp?
...
+        return lines[8].include?("Automatically created by Devel::PPPort")
```
Comment (lildude) [inline lib/linguist/generated.rb]: Will this _always_ be on line 9?

Comment (ioanrogers) [reply lib/linguist/generated.rb]: AFAICT, yes
https://github.com/mhx/Devel-PPPort/blame/d7849d96015b66d69606488aa65877dc2ba737a4/PPPort_pm.PL#L585

The git history goes back to 2012.

Comment (lildude) [reply lib/linguist/generated.rb]: I think in that case we can probably skip the `lines.count > 10` check.

Comment (ioanrogers) [reply lib/linguist/generated.rb]: In Perl, if you try to perform a match against something which is undefined you will get a warning.
I don't know Ruby - does the `include` call handle array elements which don't exist?

Comment (ioanrogers) [reply lib/linguist/generated.rb]: Okay, I just checked with a truncated file and it seemed ok.

Comment (lildude) [reply lib/linguist/generated.rb]: Ah, good point. 👍

File: lib/linguist/generated.rb (hunk 1 above)
Comment (pchaigno) [reply lib/linguist/generated.rb]: Ah, ok. I guess I was expecting `> 9` or a higher number for the whole header comment.

**Comment by lildude** [comment]
This is looking good from my perspective, except for the heeeaaawwg `ppport.h` file. As this is only being used for testing, I think we can safely truncate it to about 30 lines with a comment at the point of truncation along the lines of "linguist: deliberately truncated for test purposes" or something along those lines.

Thoughts?

**Comment by ioanrogers** [comment]
Done

**Pull request merged**